components/tikv_util/src/metrics/mod.rs

```diff
@@ -31,21 +34,22 @@ pub use self::metrics_reader::HistogramReader;
 
 mod metrics_reader;
 
-use std::collections::HashMap;
-
-use kvproto::pdpb;
 pub type RecordPairVec = Vec<pdpb::RecordPair>;
 
 pub fn dump() -> String {
     let mut buffer = vec![];
+    dump_to(&mut buffer);
+    String::from_utf8(buffer).unwrap()
```

Comment:
Is there twice mem copy?

Reply:
No. the buffer is moved into the string.